packages/SwingSet/docs/host-app.md

```diff
@@ -0,0 +1,59 @@
+# The Host Application
+
+SwingSet is a library that helps you write a "Host Application" around a kernel, which in turn manages some number of vats and devices. The Host Application does not come with SwingSet: you must write one specific to your particular application.
+
+## Host Application Responsibilities
+
+### State Management
+
+First, the host app is responsible for managing the kernel's state. It should do this by creating a "swing store", using the `@agoric/swing-store` package, which is a SQLite database in some user-selected base directory, wrapped in useful APIs. When you create the swingstore, you get back both a `hostStorage` facet and a `kernelStorage` facet. The `kernelStorage` facet must be given to `makeSwingsetController()`, the primary API for creating a kernel. All kernel state is kept inside the swing store.
+
+The host app must use `hostStorage.commit()` to commit the swingstore changes after each group of device inputs and cranks (usually triggered with a call to `controller.run()`.
+
+### Device IO
+
+The host app is also responsible for all device IO. The kernel itself cannot talk to the outside world, except through devices. These devices are configured with the kernel config record (along with static vats), but the device *endowments* are provided by the host app via the `deviceEndowments` argument to `makeSwingsetController()`.
+
+SwingSet provides robust and deterministic computation, even in the face of unexpected reboot, and avoids a failure mode called "hangover inconsistency" by following the lead of the Waterken and E systems. Output messages must be embargoed until all consequences of an inbound delivery have been durably committed. To maintain this, device endowments must refrain from transmitting their outputs until the host app calls `hostStorage.commit()`, and they must be prepared to re-transmit their outputs if they awaken into a world where the durable state says that a message must be transmitted but no record of an acknowledgment is also recorded. See the comms subsystem, especially the "mailbox" device, for more details.
```

Comment:
I think we should generalize the devices not just to transmission but to effects.

```suggestion
SwingSet provides robust and deterministic computation, even in the face of unexpected reboot, and avoids a failure mode called "hangover inconsistency" by following the lead of the Waterken and E systems. Output messages must be embargoed until all consequences of an inbound delivery have been durably committed. To maintain this, device endowments must prevent their outputs from causing persistent effects until after the host app calls `hostStorage.commit()`, and they must be prepared to reapply effects from their outputs if they awaken into a world where the durable state says that a message was processed but no trace of their effect was also recorded. See the comms subsystem, especially the "mailbox" device, for more details.
```

Reply:
I'll tweak it. I think "causing persistent effects" makes it too easy to miss the fact that revealing anything to the outside world also qualifies (readers might think that "persistent" requires a database, when a simple TCP message to an external party would be just as bad).